conf/modules.config

```
@@ -12,28 +12,42 @@
 
 process {
 
-    withName: FASTP {
+    withName: "AMPLICON_PIPELINE:READS_QC_MERGE:FASTP" {
         publishDir = [
             [
                 path: { "${params.outdir}/${meta.id}/qc/" },
                 mode: params.publish_dir_mode,
                 failOnError: true,
-                pattern: "*[!_{1,2}].fastp.fastq.gz"
+                saveAs: {
+                    filename ->
+                    {
+                        if ( meta.single_end ) {
+                            if ( filename.contains(".fastp.fastq.gz") ){
+                                return filename;
+                            }
+                            else{
+                                return null;
+                            }
+                        }
+                        else {
+                            if ( filename.contains(".merged.fastq.gz") ){
+                                return filename;
+                            }
+                            else{
+                                return null;
+                            }
```

```suggestion
if ((meta.single_end && filename.contains(".fastp.fastq.gz")) ||
    (!meta.single_end && filename.contains(".merged.fastq.gz"))) {
    return filename;
} else {
    return null;
}
```
I think that less nested code is easier to follow.